src/ops/controlflow.rs

```diff
@@ -145,6 +145,26 @@ impl StaticTag for BasicBlock {
     const TAG: OpTag = OpTag::BasicBlock;
 }
 
+impl DataflowParent for BasicBlock {
+    fn inner_signature(&self) -> FunctionType {
+        match self {
+            BasicBlock::DFB {
+                inputs,
+                other_outputs,
+                tuple_sum_rows,
+                ..
+            } => {
+                // The node outputs a TupleSum before the data outputs of the block node
+                let tuple_sum_type = Type::new_tuple_sum(tuple_sum_rows.clone());
+                let mut node_outputs = vec![tuple_sum_type];
+                node_outputs.extend_from_slice(other_outputs);
+                FunctionType::new(inputs.clone(), TypeRow::from(node_outputs))
+            }
+            BasicBlock::Exit { cfg_outputs } => FunctionType::new(type_row![], cfg_outputs.clone()),
```

Comment:
This is the biggest wart here. When does a node have an inner_signature? I thought it'd be when it had Input + Output children, but Exit does not....

I'm not sure if this justifies declaring `inner_signature(&self) -> Option<FunctionType>`. If one did that, you could even put it in `OpTrait` with a default returning None, which'd save that big switch in `get_function_type`. What do you think?

Reply:
I had the same conflict when doing it - I think my preference is actually to break up the basic block enum and move the shared functionality to a block trait.

Happy to do as a follow up or here

Reply:
Yes, I think that's an improvement, and there are some other improvements that enables, so we could go that way. We could (also, independently) just drop the trait itself, as we never use it (i.e. as a bound).

I do still wonder about the default-None, though - the approach here means that for each relevant OpType we have to *both* implement the trait *and* add an extra case to `get_function_type`, whereas by defaulting to None you only have to say that an OpType is/has a function-type in one place...